packages/plugin-twitter/src/actions/post.ts

```diff
@@ -0,0 +1,200 @@
+import {
+    Action,
+    IAgentRuntime,
+    Memory,
+    State,
+    composeContext,
+    elizaLogger,
+    generateText,
+    ModelClass,
+    formatMessages,
+} from "@elizaos/core";
+import { Scraper } from "agent-twitter-client";
+
+async function composeTweet(
+    runtime: IAgentRuntime,
+    message: Memory,
+    state?: State
+): Promise<string> {
+    try {
+        // Get recent conversation history
+        const recentMessages = await runtime.messageManager.getMemories({
+            roomId: message.roomId,
+            count: 5,
+        });
+
+        const formattedHistory = formatMessages({
+            messages: recentMessages,
+            actors: state?.actorsData,
+        });
+
+        // Template for generating the tweet
+        const tweetTemplate = `
+# Context
+Recent conversation:
+${formattedHistory}
+
+Character style:
+${runtime.character.style.post.join("\n")}
+
+Topics of expertise:
+${runtime.character.topics.join(", ")}
+
+# Task
+Generate a tweet that:
+1. Relates to the recent conversation or requested topic
+2. Matches the character's style and voice
+3. Is concise and engaging
+4. Must be UNDER 180 characters (this is a strict requirement)
+5. Speaks from the perspective of ${runtime.character.name}
+
+Generate only the tweet text, no other commentary.`;
+
+        const context = await composeContext({
+            state,
+            template: tweetTemplate,
+        });
+
+        const tweetContent = await generateText({
```

Comment:
Please use generate object which allows z objects and type safe approach to this which allows letting a character maxinum

Reply:
Also please create a types file. Please review _examples/plugins for how best to write an plugin / action

Reply:
I'll review this before opening any new PRs next time. Thanks for the suggestion.